src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -0,0 +1,161 @@
+   * @param {*} selectedPath the currently selected path
+   */
+  static keysToItem(item, selectedPath) {
+    let paths = [];
+    if (item.childItems) {
+      item.childItems.some((childItem) => {
+        if (selectedPath === childItem.path) {
+          paths = [item.path];
+          // if found bail early.
+          return true;
+        }
+
+        const childPaths = CollapsingNavigationMenu.keysToItem(childItem, selectedPath);
+        if (childPaths.length > 0) {
+          paths = childPaths.concat([item.path]);
+          // if found bail early.
+          return true;
+        }
+        return false;
+      });
+    }
+
+    return paths;
+  }
+
+  /**
+   * Returns an object containing the keys of the items to open to reveal the selected path in the tree.
+   * @param {*} menuItems list of all menu items
+   * @param {*} selectedPath the currently selected path
+   */
+  static openKeysToItem(menuItems, selectedPath) {
+    return CollapsingNavigationMenu.keysToItem(menuItems, selectedPath).reduce((acc, path) => {
+      acc[path] = true;
+      return acc;
+    }, {});
+  }
+
+  static getDerivedStateFromProps({ menuItems, selectedPath }, state) {
+    const newState = {};
+    if (state.previousSelectedPath !== selectedPath) {
+      newState.openKeys = Object.assign({}, state.openKeys, CollapsingNavigationMenu.openKeysToItem(menuItems[0], selectedPath));
+      newState.previousSelectedPath = selectedPath;
+    }
+    return newState;
+  }
+
+  constructor(props) {
+    super(props);
+    const { menuItems, selectedPath } = props;
+    this.renderMenuItems = this.renderMenuItems.bind(this);
+    this.handleOnClick = this.handleOnClick.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+
+    this.state = {
+      previousSelectedPath: selectedPath,
+      openKeys: CollapsingNavigationMenu.openKeysToItem(menuItems[0], selectedPath),
+    };
+  }
+
+  handleKeyDown(event, item) {
+    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
+      this.handleOnClick(event, item);
+    }
+  }
+
+  handleOnClick(event, item) {
+    const { onSelect } = this.props;
+    const { openKeys } = this.state;
+
+    if (!item.childItems) {
+      onSelect(item.path);
+      return;
+    }
+    openKeys[item.path] = !openKeys[item.path];
+    this.setState({ openKeys });
+  }
+
+  renderMenuItems(menuItems, firstLevel) {
+    const { selectedPath } = this.props;
+    const { openKeys } = this.state;
+
+    if (!menuItems) {
+      return undefined;
+    }
+
+    return menuItems.map((item) => {
+      const itemIsOpen = openKeys[item.path];
+      const itemHasChildren = item.childItems;
+
+      return (
+        <React.Fragment key={item.path}>
+          <div className={!firstLevel ? cx('indent') : null}>
+            <div
+              className={cx(['item', { 'is-selected': selectedPath === item.path }])}
+              tabIndex="0"
+              role="button"
```

Comment:
You could use `itemHasChildren` with `aria-haspopup` to help screen readers know there is nested content that will be displayed when interacting with this button similar to what we did here: https://github.com/cerner/terra-framework/pull/766/files